src/Jackett.Common/Indexers/CardigannIndexer.cs

```diff
@@ -1478,17 +1478,14 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                                             break;
                                         case "category":
                                             var cats = MapTrackerCatToNewznab(value);
-                                            if (release.Category == null)
+                                            if (cats.Any())
                                             {
-                                                release.Category = cats;
-                                            }
-                                            else
-                                            {
-                                                foreach (var cat in cats)
-                                                {
-                                                    if (!release.Category.Contains(cat))
-                                                        release.Category.Add(cat);
-                                                }
+                                                if (release.Category == null || FieldModifiers.Contains("noappend"))
+                                                    release.Category = cats;
+                                                else
+                                                    foreach (var cat in cats)
+                                                        if (!release.Category.Contains(cat))
+                                                            release.Category.Add(cat);
```

Comment:
We can get rid of this type of foreach loop by using Union() in Linq, which does the same thing. https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.union?view=netframework-4.8

```suggestion
                                                    release.Category = release.Category.Union(cats).ToList();
```

Reply:
This is useful for performance because `List<T>.Contains(T)` runs in O(n) time, so the code as is currently runs in O(n^2).

Union, according to [the source](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq/src/System/Linq/Union.cs), uses a stripped down version of HashSet, to do the same thing. Meaning checking for distinct values is an O(1) operation and brings the entire operation speed to O(n).

Reply:
Changed.

Maybe I'm wrong but they are converting the first iterable into a HashSet and then performing the search. The search is O(1) but the conversion takes time and memory. If the lists are small, like in this case, this solution can be slower that searching in the list O(n) because n=1 in 99% of the cases.

Reply:
Reading the reference source, in this specific usage, it basically adds each item from both lists to a single HashSet, and then memcopy's the internal array right back I don't think it gets much faster than that without changing the types to a HashSet like you said, in exchange for temporarily having 2 copies of the completed list in memory at one time.

I know not to go mad over optimizing the heck out of everything, but I wanted to point out this solution now, because I've seen other places where this pattern is used that end up with a lot more entries than ~1. The internal striped down HashSet, plus lazy eval, can end up making a huge difference. And fun fact, the source code points shows this is heavily optimized to Union-ize up to 2^32-2 lists into a single set at the end.